src/components/Header/Header.tsx

```diff
@@ -2,10 +2,40 @@ import {Link} from "react-router-dom";
 import headerLogo from '../../images/header_logo.png';
 import {Button, Input} from 'antd';
 import css from './header.module.css';
+import {useCallback, useEffect, useState} from "react";
+import {actions} from "../../store/products/reducer";
+import debounce from 'lodash/debounce';
+import {useAppDispatch} from "../../hooks/useAppDispatch";
+import {getLogin} from "../../store/login/selector";
+import {useSelector} from "react-redux";
 
 const {Search} = Input
 
+interface Params {
+    text: string;
+}
+
 export const Header = () => {
+    const [params, setParams] = useState({
+        text: ""
+    })
+    const isAuth = useSelector(getLogin)
+    const dispatch = useAppDispatch()
+    const updateParams = (nextParams: Partial<Params>) => {
+        setParams((prevParams) => ({...prevParams, ...nextParams}));
+    };
+    const fetchGetProductsDebounce = useCallback(debounce((params: Params): void =>
+    dispatch(actions.fetchProducts(params) as any), 2_000), [dispatch])
+
+    const handlerButton = () => {
```

Comment:
называет хендлеры по сути выполняемое логики. handlerButton - слишком абстрактно, подходит вообще для любой кнопки
```suggestion
    const logoutHandler = () => {
```